src/workers/tx_sender.rs

```diff
@@ -72,107 +120,49 @@ impl TxSender {
         tx_send_interval: Duration,
         postgres_send: Option<PostgresMpscSend>,
     ) -> JoinHandle<anyhow::Result<()>> {
+        let (batch_send, batch_recv) = async_channel::unbounded();
+
+        for _i in 0..5 {
+            let this = self.clone();
+            let batch_recv = batch_recv.clone();
+            let postgres_send = postgres_send.clone();
+
+            tokio::spawn(async move {
+                while let Ok((sigs_and_slots, txs)) = batch_recv.recv().await {
+                    this.forward_txs(sigs_and_slots, txs, postgres_send.clone())
+                        .await;
+                }
+            });
+        }
+
         tokio::spawn(async move {
             info!(
-                "Batching tx(s) with batch size of {tx_batch_size} every {} ms",
+                "Batching tx(s) with batch size of {tx_batch_size} every {}ms",
                 tx_send_interval.as_millis()
             );
 
             loop {
                 let mut sigs_and_slots = Vec::with_capacity(tx_batch_size);
                 let mut txs = Vec::with_capacity(tx_batch_size);
-                let mut maybe_permit = None;
-                let counting_semaphore = self.counting_semaphore.clone();
+
                 while txs.len() <= tx_batch_size {
-                    let res = tokio::time::timeout(tx_send_interval, recv.recv()).await;
-                    match res {
-                        Ok(value) => match value {
-                            Some((sig, tx, slot)) => {
-                                sigs_and_slots.push((sig, slot));
-                                txs.push(tx);
-                            }
-                            None => {
-                                bail!("Channel Disconnected");
-                            }
-                        },
+                    match recv.try_recv() {
```

Comment:
Please have the same logic as before with tokio::time::timeout etc.

Reply:
Fixed